go-sqlsmith/stateflow/schema.go

```diff
@@ -59,22 +59,34 @@ func (s *StateFlow) randTableFromTable(table *types.Table, newName bool, fn bool
 	return &newTable
 }
 
-func (s *StateFlow) randTable(newName bool, fn bool) (*types.Table) {
-	newTable := new(types.Table)
-	tables := s.db.Tables
-	index := 0
-	k := util.Rd(len(tables))
-	for _, table := range tables {
-		if index == k {
-			for len(newTable.Columns) == 0 {
-				newTable = s.randTableFromTable(table, newName, fn)
-			}
-			return newTable
+func (s *StateFlow) randTable(newName bool, fn bool, online bool) (*types.Table) {
+	tables := []*types.Table{}
+	for _, table := range s.db.Tables {
+		// get online tables
+		if !online || !s.db.Online || table.Online {
```

Comment:
I'm confused, db.Online and table.Online wasn't always set to same boolean value? 😶

Reply:
They are not, this is for avoiding online DDL, when start a txn, db.Online and some tables will be set to true, and the other tables has Online value false, then the go-sqlsmith should generate DDL SQLs only about these tables during this txn.